dbaccess/block.go

```diff
@@ -0,0 +1,109 @@
+package dbaccess
+
+import (
+	"github.com/kaspanet/kaspad/database2"
+	"github.com/kaspanet/kaspad/util"
+	"github.com/kaspanet/kaspad/util/daghash"
+	"github.com/pkg/errors"
+)
+
+const (
+	blockStoreName = "blocks"
+)
+
+var (
+	blockLocationsBucket = database2.MakeBucket([]byte("block-locations"))
+)
+
+// StoreBlock stores the given block in the database.
+func StoreBlock(context Context, block *util.Block) error {
+	db, err := context.db()
+	if err != nil {
+		return err
+	}
+
+	// Make sure that the block does not already exist.
+	hash := block.Hash()
+	exists, err := HasBlock(context, hash)
+	if err != nil {
+		return err
+	}
+	if exists {
+		return errors.Errorf("block %s already exists", hash)
+	}
+
+	// Save a reference to the current block location in case
+	// we fail and need to rollback.
+	previousBlockLocation := db.CurrentStoreLocation(blockStoreName)
+	rollback := func() error {
+		return db.RollbackStore(blockStoreName, previousBlockLocation)
+	}
+
+	// Write the block's bytes to the block store and rollback
+	// if there's an error.
+	bytes, err := block.Bytes()
+	if err != nil {
+		return err
+	}
+	blockLocation, err := db.AppendToStore(blockStoreName, bytes)
+	if err != nil {
+		rollbackErr := rollback()
+		if rollbackErr != nil {
+			return rollbackErr
```

Comment:
The original error is more important than the rollback error.
Try to somehow combine if possible (Shouldn't be too hard I think), but if not - return the original error